target/ppc/translate.c

```diff
@@ -7457,6 +7457,15 @@ static int times_4(DisasContext *ctx, int x)
 # define REQUIRE_64BIT(CTX)  REQUIRE_INSNS_FLAGS(CTX, 64B)
 #endif
 
+#define VOID
```

Comment:
Personally, I don't like the idea of #defining VOID to nothing just so you can reuse REQUIRE_VECTOR in the legacy code. Additionally, although clever, this approach makes REQUIRE_VECTOR more complex than it needs to be, and users of REQUIRE_VECTOR would need to care about which value to return, instead of trusting it to do the right thing (which I expect it would, for any new decodetree-based instruction implementations).

An alternative would be to define a separate REQUIRE_VECTOR_LEGACY(CTX) to be used in the legacy implementation, but I think I'd rather just keep the legacy code as is, and use REQUIRE_VECTOR just for the new instructions. Eventually, the whole legacy implementation should be replaced.

Reply:
That's fair. I wanted to implement mferst's idea of replacing everything and for that I'd need to change it like this (I tried using void without the define, but didn't work =/ ) I guess only using it on new code could be the better approach. Decode tree should always return true?

Reply:
> Decode tree should always return true?

I believe so, as this macro would be used in the trans_* functions. Returning true from trans_* means the instruction translation was properly handled, which, in this case, means emitting code to generate the VPU exception.